Sprint_3/source/CardGames.py

```diff
@@ -0,0 +1,368 @@
+      return False
+    for player in players:
+      for _ in range(numCards):
+        player.addCard(self.deck.getCard())
+    return True
+  
+  def addCardsToDealer(self, amount=1):
+    for _ in range(amount):
+      if len(self.deck.cards) > 0:
+        self.hand.append(self.deck.getCard())
+      else:
+        print("No more cards left in deck to add.")
+        break
+
+  def resetDeck(self):
+    self.deck.reset()
+    self.deck.shuffle()
+
+
+# has_pair function---------------------------------------------------------------
+
+def has_pair(player):
+    Hand = player.hand
+    values = [card.value for card in Hand]
+    compare = set()
+    print(values, compare)
+    for i in values:
+        if i in compare:
+            return True
+        compare.add(i)
+    else:
+        return False
+#---------------------------------------------------------------------------------
+
+
+# call function--------------------------------------------------------------
+
+def Call(player, bet: int, pot):
+    player.makeBet(bet, pot)
+
+
+
+def highest_card(hand):
+    # Check if the hand is empty
+      if not hand:
+          return None
+
+    # Find the card with the highest value
+      highest_card = max(hand, key=lambda card: card.value)
+    
+    # Return the highest card
+      return highest_card
+
+
+class Poker:
+  def __init__(self, players: "list[Player]", dealer: Dealer):
+    self.players = players
+    self.dealer = dealer
+    self.pot = 0
+    self.bets = [0 for _ in range(len(players))]
+    self.round = 0
+    self.currentPlayer = 0
+
+
+  # Add all functions pertaining to the game of poker below
+  # Add global variables to the __init__ function above
+
+  # Keep Track of Pot, Bets, and current round
+  def addMoneyToPot(self, amount: int):
+    self.pot += amount
+    return self.pot
+  
+  def addBet(self, player: Player, amount: int):
+    idx = self.players.index(player)
+    self.bets[idx] += amount
+    return self.bets[idx]
+  
+  # Add all bets to the pot
+  def collectBets(self):
+    for bet in self.bets:
+      self.addMoneyToPot(bet)
+    self.bets = [0 for _ in range(len(self.players))]
+
+  # Check if all bets are equal
+  def checkBets(self):
+    return len(set(self.bets)) == 1
+  
+  # Check how much is needed to call
+  def callAmount(self, player: Player):
+    idx = self.players.index(player)
+    return max(self.bets) - self.bets[idx]
+  
+class Identification: 
+  
+  def __init__(self, hand):
+    self.hand = hand
+    self.name = ""
+    
+  def three_of_kind(self):            
+      # Get all the rank values in the given hand.         
+    values = [card.value for card in self.hand]
+      # Count occurences of each value.
+    val_count = []
+    for v in set(values): val_count.append(values.count(v))
+    val_count.sort()
+      # Identify as "Three of a Kind": 3 Cards same rank + 2 unrelated cards.
+    if val_count == [1, 1, 3]:
+      self.name = "Three of a Kind"
```

Comment:
You will also want to check for cases of 3 and 2 if there were three of a kind and two of a kind. Other than that everything looks good!